chips/nrf52/src/chip.rs

```diff
@@ -12,23 +12,101 @@ use kernel::common::deferred_call;
 use kernel::debug;
 use nrf5x::peripheral_interrupts;
 
+pub trait InterruptServiceTrait {
```

Comment:
```suggestion
pub trait InterruptService {
```

Reply:
Although I agree that the current naming is not ideal, I'm not convinced that `InterruptService`/`InterruptServicer` is clearer than `InterruptServiceTrait`/`InterruptService`.

If the trait moves to a separate module (which would ultimately be its place - e.g. in the kernel), then everything could be an `InterruptService`, and the trait would appear in this file as `kernel::InterruptService` or something like that. Not sure that's the clearest either.

Another possibility is to name the implementations `Nrf52InterruptService`/`Nrf52840InterruptService`, although this can be slightly redundant with the module they're defined in (i.e. `nrf52::Nrf52InterruptService` or `nrf52840::Nrf52840InterruptService`).

WDYT? Any other idea?

Reply:
I think naming a trait with Trait is redundant, and isn't used elsewhere in Tock. I do like being specific with the chip-specific naming: Nrf52InterruptService/Nrf52840InterruptService.